src/main/kotlin/com/terraformation/backend/search/table/DocumentsTable.kt

```diff
@@ -0,0 +1,56 @@
+package com.terraformation.backend.search.table
+
+import com.terraformation.backend.auth.currentUser
+import com.terraformation.backend.db.default_schema.ProjectId
+import com.terraformation.backend.db.default_schema.tables.references.PROJECTS
+import com.terraformation.backend.db.docprod.DocumentId
+import com.terraformation.backend.db.docprod.DocumentTemplateId
+import com.terraformation.backend.db.docprod.tables.references.DOCUMENTS
+import com.terraformation.backend.db.docprod.tables.references.DOCUMENT_SAVED_VERSIONS
+import com.terraformation.backend.search.SearchTable
+import com.terraformation.backend.search.SublistField
+import com.terraformation.backend.search.field.SearchField
+import org.jooq.Condition
+import org.jooq.OrderField
+import org.jooq.Record
+import org.jooq.TableField
+import org.jooq.impl.DSL
+import org.jooq.impl.SQLDataType
+
+class DocumentsTable(tables: SearchTables) : SearchTable() {
+  override val primaryKey: TableField<out Record, out Any?>
+    get() = DOCUMENTS.ID
+
+  override val sublists: List<SublistField> by lazy {
+    with(tables) {
+      listOf(projects.asMultiValueSublist("projects", PROJECTS.ID.eq(DOCUMENTS.PROJECT_ID)))
```

Comment:
I think this should be a single-value sublist since a document can only have one project. (And it should be named `project` instead of `projects` to reflect that.)

Also, the projects table probably ought to have a multi-value sublist pointing to this table. Which would also eliminate the need to add special handling of prefixes; you'd be able to address this table with a prefix of `projects.documents`.